src/transformers/models/llama/modeling_llama.py

```diff
@@ -172,10 +172,18 @@ def __init__(
         self.config = config
         self.rope_init_fn = ROPE_INIT_FUNCTIONS[self.rope_type]
 
-        inv_freq, self.attention_scaling = self.rope_init_fn(self.config, device, **self.rope_kwargs)
-        self.register_buffer("inv_freq", inv_freq, persistent=False)
+        # Init function for inv_freq
+        def init_inv_freq(device: torch.device):
+            inv_freq, self.attention_scaling = self.rope_init_fn(self.config, device, **self.rope_kwargs)
+            return inv_freq
+
+        self.register_buffer("inv_freq", init_inv_freq(device), persistent=False)
         self.original_inv_freq = self.inv_freq
 
+        # Save buffer init callback
+        LlamaModel.buf_init_callbacks.setdefault("rotary_emb.inv_freq", init_inv_freq)
```

Comment:
can this not be called in the `LlamaModel` directly, it's a bit weird for us to register something like this at the class level